token import and export

jona-samuel opened the pull request:
New im-/export for tokens
- [x] 4Eyes
- [x] ApplicationSpecificPassword
- [x] Certificate
- [x] DayPassword
- [x] EMail
- [x] HOTP
- [x] Indexed Secret
~mOTP~
- [x] OCRA
- [x] Paper
- [x] Push
- [x] Password
- [x] Questionnaire
- [x] Registration
~RADIUS~
~Remote~ 
- [x] SMS
- [x] Spass
- [x] SSH
- [x] TAN
- [x] TiQR
- [x] TOTP
~U2F~
~VASCO~
- [x] WebAuthn
- [x] Yubikey
~Yubico~

jona-samuel committed:
new im-/export functions

jona-samuel committed:
rebase

jona-samuel committed:
im-/export hotp

jona-samuel committed:
im-/export hotp

jona-samuel committed:
im-/export totp

jona-samuel committed:
im-/export totp

jona-samuel committed:
im-/export email

jona-samuel committed:
im-/export email

jona-samuel committed:
im-/export sms and small fixes

jona-samuel committed:
im-/export sms and small fixes

jona-samuel committed:
clean up

jona-samuel committed:
clean up

nilsbehlen committed:
update formatting

nilsbehlen committed:
update formatting

nilsbehlen committed:
make result immutable

nilsbehlen committed:
make result immutable

jona-samuel committed:
type hints

jona-samuel committed:
type hints

jona-samuel committed:
4eyes im-/export

jona-samuel committed:
4eyes im-/export

jona-samuel committed:
Merge branch 'master' into 4000/Token-import-and-export

jona-samuel committed:
push im-/export

jona-samuel committed:
push im-/export

jona-samuel committed:
fixing test

jona-samuel committed:
fixing test

jona-samuel committed:
adding test

jona-samuel committed:
adding test

jona-samuel commented:
First tokens for #4342

nilsbehlen commented on privacyidea/lib/token.py:
```diff
@@ -58,8 +58,11 @@
 This is the middleware/glue between the HTTP API and the database
 """
+
+from dataclasses import dataclass
```
sorting of imports, try to use "optimize imports" of pycharm

nilsbehlen commented on privacyidea/lib/tokens/smstoken.py:
```diff
@@ -631,3 +631,25 @@ def enroll_via_validate_2nd_step(self, passw, options=None):
+        if not token_information["dynamic_phone"]:
+            self.add_tokeninfo("phone", token_information["phone"])
+        self.save()
+        pass
```
+1

nilsbehlen commented on privacyidea/lib/tokens/pushtoken.py:
```diff
@@ -1211,3 +1211,29 @@ def get_enroll_url(self, user: User, params: dict) -> str:
+        Import a push token.
+        """
+        TokenClass.import_token(self, token_information)
+        self.add_tokeninfo("public_key_smartphone", token_information["public_key_smartphone"])
```
better use `add_tokeninfo_dict` to only have one database operation for all the tokeninfo

jona-samuel committed:
requested changes

jona-samuel committed:
requested changes

jona-samuel committed:
bug fix

jona-samuel committed:
bug fix

jona-samuel committed:
improvements

jona-samuel committed:
improvements

jona-samuel committed:
fixing test

jona-samuel committed:
fixing test

jona-samuel committed:
implementing requested changes

jona-samuel committed:
implementing requested changes

nilsbehlen commented on privacyidea/lib/tokenclass.py:
```diff
@@ -2007,3 +2007,31 @@ def get_enroll_url(self, user: User, params: dict):
+            self.token.description = token_information["description"]
+            self.add_tokeninfo_dict(token_information["tokeninfo"])
+            self.add_tokeninfo("improrted", datetime.datetime.now(datetime.timezone.utc).isoformat(timespec="seconds"))
+            self.save()
```
"import_date"

jona-samuel committed:
adding webauthn, passkey und yubikey

jona-samuel committed:
adding webauthn, passkey und yubikey

jona-samuel committed:
adding indexedsecret, tan and tiqr

jona-samuel committed:
adding indexedsecret, tan and tiqr

jona-samuel committed:
fixing test

jona-samuel committed:
fixing test

jona-samuel committed:
Merge branch 'master' into 4000/Token-import-and-export

jona-samuel committed:
adding ssh-, questionnaire- and registrationtoken

jona-samuel committed:
adding ssh-, questionnaire- and registrationtoken

jona-samuel committed:
fixing test

jona-samuel committed:
fixing test

jona-samuel committed:
Codecov and skipping not supported tokens

jona-samuel committed:
Codecov and skipping not supported tokens

jona-samuel committed:
Remove old unused code

jona-samuel committed:
Remove old unused code

jona-samuel committed:
Fixing test

jona-samuel committed:
Fixing test

jona-samuel committed:
Fixing test

jona-samuel committed:
Fixing test

jona-samuel committed:
Fixing test more tests

jona-samuel committed:
Fixing test more tests

jona-samuel committed:
Clean up

jona-samuel committed:
Clean up

plettich committed:
Split `modules.py` into smaller units

Remove the `mysql_row_format` table parameter since `DYNAMIC` is the
default for mysql for quite some time now.

Improve some spelling, formatting and coverage.

Closes: #4468

jona-samuel committed:
improvement

jona-samuel committed:
improvement

nilsbehlen committed:
Update caconnector.py

jona-samuel committed:
Merge branch 'master' into 4000/Token-import-and-export

nilsbehlen committed:
Merge pull request #4471 from privacyidea/4468/split_models_into_smaller_units

Split `modules.py` into smaller units

jona-samuel committed:
resolve merge conflict

jona-samuel committed:
Merge remote-tracking branch 'origin/4000/Token-import-and-export' into 4000/Token-import-and-export

# Conflicts:
#	privacyidea/models.py
#	tests/test_lib_tokens_foureyes.py
#	tests/test_lib_tokens_radius.py
#	tests/test_lib_tokens_spass.py
#	tests/test_lib_tokens_webauthn.py

jona-samuel commented:
Replace this with a new pull request to fix merch problems
